toDoApp/app/src/main/java/hieunguyen/com/todoapp/MainActivity.java

```diff
@@ -0,0 +1,181 @@
+
+        // On tap, allow user to edit task
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                createEditDialog(position).show();
+            }
+        });
+
+        // On long hold, allow user to delete task
+        listView.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+            @Override
+            public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+                createDeleteDialog(position).show();
+                return true;
+            }
+        });
+
+        Button clearCompleted = (Button) findViewById(R.id.clearCompleted);
+        Button addTask = (Button) findViewById(R.id.addNew);
+
+        // Remove completed tasks
+        clearCompleted.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                dbService.clearCompleted();
+                itemsAdapter.setList(dbService.getDataFromDB());
+                updateListHeader();
+            }
+        });
+
+        // On tap, allow user to add task
+        addTask.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createAddDialog().show();
+            }
+        });
+    }
+
+    /**
+     * Updates the header that says "X To Do"
+     */
+    public void updateListHeader() {
+        TextView header = (TextView) findViewById(R.id.header);
+        String count = itemsAdapter.getCount() == 0 ? "Nice! Nothing" : String.valueOf(itemsAdapter.getCount());
+        header.setText(count + " To Do");
+    }
+
+    /**
+     * Creates a pop-up to allow user to edit the note
+     * @param position position of note being edited in the ArrayAdapter
+     * @return AlertDialog.Builder object
+     */
+    public AlertDialog createEditDialog(final int position) {
+        String currentItem = itemsAdapter.getList().get(position).getText();
+
+        AlertDialog.Builder editDialogBuilder = new AlertDialog.Builder(MainActivity.this);
+
+        final EditText editText = new EditText(MainActivity.this);
+        editText.setText(currentItem);
+        editText.setSelection(currentItem.length());
+
+        editDialogBuilder.setPositiveButton(R.string.confirmEdit, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                ToDoItem task = itemsAdapter.getList().get(position);
+                task.setText(editText.getText().toString());
+                dbService.edit(task);
+                itemsAdapter.setList(dbService.getDataFromDB());
+                updateListHeader();
+            }
+        }).setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
```

Comment:
Instead of creating on OnClickListener that does nothing, you can pass null, and the alertdialog will still be dismissed, I believe